add autologging ignore option

naseemkullah opened the pull request:
solves #107 via a generic `autologging.ignore` function 

I get errors locally when trying to test: `Unexpected var, use let or const instead.`, linting fixes it (`npm run fix`) but have reverted the linting as requested

naseemkullah commented on logger.js:
```diff
@@ -38,39 +38,40 @@ function pinoLogger (opts, stream) {
+  const autoLogging = (opts.autoLogging !== false)
+  const autoLoggingIgnorePaths = (opts.autoLogging && opts.autoLogging.ignorePaths) ? opts.autoLogging.ignorePaths : []
+  const autoLoggingGetPath = opts.autoLogging && opts.autoLogging.getPath ? opts.autoLogging.getPath : null
+  const autoLoggingIgnore = opts.autoLogging && opts.autoLogging.ignore ? opts.autoLogging.ignore : null
```
here it starts

naseemkullah commented on logger.js:
```diff
@@ -87,34 +88,41 @@ function pinoLogger (opts, stream) {
 
           return ignorePath === url.pathname
         })
+
+        shouldLogSuccess = !isPathIgnored
+      }
+
+      if (autoLoggingIgnore !== null && shouldLogSuccess === true) {
+        const isIgnored = autoLoggingIgnore !== null && autoLoggingIgnore(req)
+        shouldLogSuccess = !isIgnored
```
and then here

naseemkullah commented on test.js:
```diff
@@ -401,9 +401,38 @@ test('no auto logging with autoLogging set to use regular expressions. result is
   })
 })
 
+test('no auto logging with autoLogging set to true and ignoring a specific user-agent', function (t) {
```
test here

mcollina reviewed (changes requested):
Can you please restore all the spurious changes? There is a lot of var -> const/let migration here.

naseemkullah commented:
> Can you please restore all the spurious changes? There is a lot of var -> const/let migration here.

linting reverted, but tests fail locally (and in CI)

mcollina reviewed (approved):
lgtm

mcollina commented:
CI is failing because:

```
  /home/runner/work/pino-http/pino-http/logger.js:122:11: 'autoLoggingIgnore' is not defined.
61
  /home/runner/work/pino-http/pino-http/logger.js:123:27: 'autoLoggingIgnore' is not defined.
62
  /home/runner/work/pino-http/pino-http/logger.js:123:57: 'autoLoggingIgnore' is not defined.
```

naseemkullah commented:
> CI is failing because:
> 
> ```
>   /home/runner/work/pino-http/pino-http/logger.js:122:11: 'autoLoggingIgnore' is not defined.
> 61
>   /home/runner/work/pino-http/pino-http/logger.js:123:27: 'autoLoggingIgnore' is not defined.
> 62
>   /home/runner/work/pino-http/pino-http/logger.js:123:57: 'autoLoggingIgnore' is not defined.
> ```

Oops thanks, removed it when i reverted the linting, added back.

> CI is failing because:
> 
> ```
>   /home/runner/work/pino-http/pino-http/logger.js:122:11: 'autoLoggingIgnore' is not defined.
> 61
>   /home/runner/work/pino-http/pino-http/logger.js:123:27: 'autoLoggingIgnore' is not defined.
> 62
>   /home/runner/work/pino-http/pino-http/logger.js:123:57: 'autoLoggingIgnore' is not defined.
> ```

Oops, thanks. Fixing now.
